src/components/structures/login/Login.js

```diff
@@ -432,8 +572,8 @@ module.exports = React.createClass({
         if (!SdkConfig.get()['disable_custom_urls']) {
             serverConfig = <ServerConfig ref="serverConfig"
                 withToggleButton={true}
-                customHsUrl={this.props.customHsUrl}
-                customIsUrl={this.props.customIsUrl}
+                customHsUrl={this.state.discoveredHsUrl || this.props.customHsUrl}
+                customIsUrl={this.state.discoveredIsUrl ||this.props.customIsUrl}
```

Comment:
Should the value from .well-known be changing the default rather than setting custom URLs? You may still want to override the HS URL (eg. `https://example.com/` -> `https://example-test.com/`) but having got the urls from .well-known essentially provides us a better default.

Reply:
There was something that got in the way when I tried to set them explicitly the first time, however I think I might have fixed the bug towards the end of implementing this. I'll see if I can repurpose the custom properties instead of introducing new ones, while also maintaining the ability to override the discovered URL (for power users).

Reply:
I think it's best to keep this as-is. It forces the UI to switch to "Custom Server", signifying that something happened. It isn't the best UX, but it is something.